src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -796,12 +787,15 @@ result::UpdateCheck SotaUptaneClient::checkUpdates() {
 
   std::vector<Uptane::Target> updates;
   unsigned int ecus_count = 0;
-  if (!uptaneIteration(&updates, &ecus_count)) {
+  try {
+    uptaneIteration(&updates, &ecus_count);
+  } catch (const std::exception &e) {
     result = result::UpdateCheck({}, 0, result::UpdateStatus::kError, Json::nullValue, "Could not update metadata.");
     return result;
   }
 
   std::string director_targets;
+  // What if it fails?
```

Comment:
It should abort, right? What else is there to do?

Reply:
Yes, we can add some check then.